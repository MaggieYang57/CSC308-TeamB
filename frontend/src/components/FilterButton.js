import React from "react";
import "../css/filterButton.css";
import { PropTypes } from 'prop-types';

function FilterButton(props) {
  function handleFilterChange() {
    props.onChange(props.filterType);
  }

  return (
    <div className="filterButton">
      <label>
        <input type="checkbox" value="1" onClick={handleFilterChange} />
        <span>{props.name}</span>
      </label>
    </div>
  );
}

FilterButton.propTypes = {
  name: PropTypes.string,
  filterType: PropTypes.string,
  onChange: PropTypes.func,
};
export default FilterButton;
