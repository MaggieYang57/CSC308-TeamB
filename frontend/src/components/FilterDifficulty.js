import React, { useState } from "react";
import "../css/filterBar.css";

function FilterDifficulty(props) {
  const [difficulty, setDifficulty] = useState("All")

  function handleFilterChange(e) {
    console.log(e.target.value);
    setDifficulty(e.target.value);
    props.onChange(e.target.value);
  }

  return (
    <div id="difficultyFilter">
      <form>
        <label for="difficulty">Filter Hikes by Difficulty:</label>
        <select name="difficulty" id="difficulty" onChange={handleFilterChange}>
          <option value="0">All</option>
          <option value="1">Easy</option>
          <option value="3">Moderate</option>
          <option value="4">Hard</option>
          <option value="5">Expert</option>
        </select>
      </form>
    </div>
  );
}

export default FilterDifficulty;
