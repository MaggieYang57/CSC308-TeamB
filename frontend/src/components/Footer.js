import React from 'react';
import { Card, Badge, Button, NavBar, Form} from 'react-bootstrap';

export function Footer({}) {
	 return (
			<div>
				<nav class="navbar navbar-dark navbar-fixed-bottom" style={{backgroundColor: '#d7e0e2'}}>
				<a class="navbar-brand">
					<form class="form-inline " style = {{marginLeft: '3vw'}} >
					<b class="text text-center " style={{fontSize: '1.2vw', color: "#2C6674"}}>SLO HIKES</b>
					<b class="text text-center " style={{fontSize: '.7vw', color: "#828282", padding: '1vw', textAlign: 'center'}}>@ SLOHikes by Sarah Rietkerk, Maggie Yang, Mike Erinberg, Bat Battur, and Skylar Kurth.</b>
					</form>
				</a>
				</nav>
			</div>
	 	)
}