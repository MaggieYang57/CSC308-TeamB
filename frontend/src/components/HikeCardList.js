import React from 'react';
import { Card, Badge, Button, Row, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';

import { Ratings } from './Ratings'

function round(value, precision) {
    var multiplier = Math.pow(10, precision || 0);
    return Math.round(value * multiplier) / multiplier;
}

export function HikeCardList({ hike, setOrdered}) {
	return (
		<Card style={{ width: '45rem', height: '13rem', margin: "auto", border: "none"}}>
		  <Row className='no-gutters'>
			  <Col md={5} lg={3}  >
			  	<Card.Img variant="top" src={hike.imagesrc} style={{height: '10rem', width: '15rem', marginTop: '1rem'}}fluid />
			  </Col>
			  <Col>
				  <Card.Body style={{marginLeft: '4rem', marginTop: '1rem'}}>
				  	<Ratings rating = {round(hike.rating, 1)} />
				    <Card.Title style={{marginTop: '1vw'}}>
							<Link to={"/hike/" + hike._id} >
							<p class=" h5 card-text text-left text-primary font-weight-bold" >{hike.title} </p>
							</Link>
				    </Card.Title>
				    <Card.Text>
				      <p class="card-text text-left" style={{fontWeight:'bold'}}>{hike.location}</p>
				    </Card.Text>

				  </Card.Body>
			  </Col>
		  </Row>
		  <Row>
		  	<Card.Text style={{marginLeft: '1rem', marginTop: '.7rem'}} >
		      <p class="card-text text-left">{hike.description.substring(0,170) + "..."}</p>
		    </Card.Text>
		  </Row>
		</Card>

   )
}
