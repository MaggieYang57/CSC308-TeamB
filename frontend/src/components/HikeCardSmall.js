import React from "react";
import { Card, Row, Col } from "react-bootstrap";
import { Link } from "react-router-dom";
import { PropTypes } from 'prop-types';

export function HikeCardList({ hike, setOrdered }) {
  return (
    <Card
      style={{
        width: "50rem",
        height: "13rem",
        margin: "auto",
        border: "none",
      }}
    >
      <Row className="no-gutters">
        <Col md={5} lg={3}>
          <Card.Img
            variant="top"
            src={"/images/default.png"}
            style={{ height: "10rem", width: "15rem", marginTop: "1rem" }}
            fluid
          />
        </Col>
        <Col>
          <Card.Body style={{ marginLeft: "4rem" }}>
            <Card.Title>
              <Link to={"/hike/" + hike._id}>
                <p className=" h5 card-text text-left text-primary font-weight-bold">
                  {hike.title}
                </p>
              </Link>
            </Card.Title>
            <Card.Text>
              <p className="card-text text-left" style={{ fontWeight: "bold" }}>
                {hike.location}
              </p>
            </Card.Text>

            <Card.Text>
              <p className="card-text text-left">
                {hike.description.substring(0, 150) + "..."}
              </p>
            </Card.Text>
          </Card.Body>
        </Col>
      </Row>
    </Card>
  );
}

HikeCardList.propTypes = {
  hike: PropTypes.object,
  setOrdered: PropTypes.object,
};