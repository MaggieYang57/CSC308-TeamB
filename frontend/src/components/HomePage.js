import React from 'react';
import { Link } from "react-router-dom";
import '../css/HomePage.css';

class HomePage extends React.Component {
  render() {
    return (
        <div className="home">
            <div className= "top-banner">
                <div className = "banner-info">
                    <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="270" />
                    <div>
                        <h1>Want to get out and explore?</h1>
                        <p>This body of text should give a little info on our site and what users can do here. Explain breifly how we can help them, why they want to stay and use our site, maybe a little bit about what our mission is. But not too much text, need to keep it light. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras sodales, diam eget tristique ultrices, orci massa consequat sapien, nec egestas sem tellus et nisi. </p>
                        <Link to="/hikeFinder"><button href="hikefinder">FIND A HIKE NOW</button></Link>
                    </div>
                </div>
            </div>
            <div className = "featured-hikes">
                <h1 id="featured-title">Most Popular Hikes</h1>
                <div className = "row">
                    <div id = "column1">
                        <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="120" />
                        <div className = "info">
                            <h1>Hike Title</h1>
                            <h2>Location</h2>
                            <p>Short description here</p>
                        </div>
                    </div>
                    <div id = "column2">
                    <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="120" />
                        <div className = "info">
                            <h1>Hike Title</h1>
                            <h2>Location</h2>
                            <p>Short description here</p>
                        </div>
                    </div>
                    <div id = "column3">
                    <img src = "https://www.margarita-adventures.com/wp-content/uploads/2017/02/Cerro_San_Luis.jpg" height="120" />
                        <div className = "info">
                            <h1>Hike Title</h1>
                            <h2>Location</h2>
                            <p>Short description here</p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
  }
}
export default HomePage;