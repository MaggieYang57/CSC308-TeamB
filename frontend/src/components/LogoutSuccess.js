import React from "react";
import { Link } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "../css/Signup.css";

function LogoutSuccess(props) {
  return (
    <div className="signup-form">
      <div id="title-signup">
        <b className="text text-center " style={{ fontSize: 40, color: "#2C6674" }}>
          Logout Filler
        </b>
      </div>
      <p style={{ marginTop: 20 }}>
        Back to <Link to="/">Homepage</Link>
      </p>
    </div>
  );
}

export default LogoutSuccess;
