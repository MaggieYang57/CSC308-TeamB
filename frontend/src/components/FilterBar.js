import React, { Component, useState, useEffect } from "react";
import axios from "axios";
import "../css/filterBar.css";
import FilterButton from "./FilterButton";
import SortDrop from "./SortDrop";
import SearchBar from "./SearchBar";

function FilterBar(props) {

  return (
    <div id="filterBar">
      <SearchBar />
      <SortDrop onChange={props.onDiff}/>
      
      <div id="filter-buttons">
        <FilterButton name="Dog Friendly" filterType='dog_friendly' onChange={props.onChange} />
        <FilterButton name="Family Friendly" filterType='family_friendly' onChange={props.onChange}/>
        <FilterButton name="Horseback Riding" filterType='horseback_riding' onChange={props.onChange}/>
        <FilterButton name="Bike Riding" filterType='biking' onChange={props.onChange}/>
      </div>
    </div>
  );
  
}

export default FilterBar;
