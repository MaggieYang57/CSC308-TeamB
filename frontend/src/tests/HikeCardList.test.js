import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router';
import { HikeCardList } from "../components/HikeCardList";

describe("HikeCardList", () => {

  it('renders correctly', () => {
    const mockHike = {
      '_id': 1, 
      'rating': [5, 3, 4], 
      'title':'Hike', 
      'location': 'SLO', 
      'description': 'great hike', 
      'difficulty': [1, 3, 5],
      'imagesrc': 'test' 
    }

    const tree = renderer
      .create(<MemoryRouter> <HikeCardList hike={mockHike} /> </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})