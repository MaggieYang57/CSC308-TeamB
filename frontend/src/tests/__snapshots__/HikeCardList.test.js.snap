// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HikeCardList renders correctly 1`] = `
Array [
  " ",
  <div
    className="card"
    style={
      Object {
        "border": "none",
        "height": "13rem",
        "margin": "auto",
        "width": "45rem",
      }
    }
  >
    <div
      className="no-gutters row"
    >
      <div
        className="col-lg-3 col-md-5"
      >
        <img
          className="card-img-top"
          fluid={true}
          src="test"
          style={
            Object {
              "height": "10rem",
              "marginTop": "1rem",
              "width": "15rem",
            }
          }
        />
      </div>
      <div
        className="col"
      >
        <div
          className="card-body"
          style={
            Object {
              "marginLeft": "4rem",
              "marginTop": "1rem",
            }
          }
        >
          <div>
            <form
              className="form-inline "
              style={
                Object {
                  "marginLeft": "0vw",
                }
              }
            >
              <a>
                 
                <img
                  className="img-fluid"
                  src="/images/stars/full.png"
                  style={
                    Object {
                      "borderRadius": 0,
                      "height": "2rem",
                      "width": "2.1rem",
                    }
                  }
                />
                 
              </a>
              <a
                style={
                  Object {
                    "marginLeft": ".3vw",
                  }
                }
              >
                 
                <img
                  className="img-fluid"
                  src="/images/stars/full.png"
                  style={
                    Object {
                      "borderRadius": 0,
                      "height": "2rem",
                      "width": "2.1rem",
                    }
                  }
                />
                 
              </a>
              <a
                style={
                  Object {
                    "marginLeft": ".3vw",
                  }
                }
              >
                 
                <img
                  className="img-fluid"
                  src="/images/stars/full.png"
                  style={
                    Object {
                      "borderRadius": 0,
                      "height": "2rem",
                      "width": "2.1rem",
                    }
                  }
                />
                 
              </a>
              <a
                style={
                  Object {
                    "marginLeft": ".3vw",
                  }
                }
              >
                 
                <img
                  className="img-fluid"
                  src="/images/stars/full.png"
                  style={
                    Object {
                      "borderRadius": 0,
                      "height": "2rem",
                      "width": "2.1rem",
                    }
                  }
                />
                 
              </a>
              <a
                style={
                  Object {
                    "marginLeft": ".3vw",
                  }
                }
              >
                 
                <img
                  className="img-fluid"
                  src="/images/stars/full.png"
                  style={
                    Object {
                      "borderRadius": 0,
                      "height": "2rem",
                      "width": "2.1rem",
                    }
                  }
                />
                 
              </a>
              <a
                style={
                  Object {
                    "fontSize": 15,
                    "marginLeft": ".5vw",
                  }
                }
              >
                 
                5
                 
              </a>
            </form>
          </div>
          <div
            className="card-title h5"
            style={
              Object {
                "marginTop": "1vw",
              }
            }
          >
            <a
              href="/hike/1"
              onClick={[Function]}
            >
              <p
                className=" h5 card-text text-left text-primary font-weight-bold"
              >
                Hike
                 
              </p>
            </a>
          </div>
          <p
            className="card-text"
          >
            <p
              className="card-text text-left"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              SLO
            </p>
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <p
        className="card-text"
        style={
          Object {
            "marginLeft": "1rem",
            "marginTop": ".7rem",
          }
        }
      >
        <p
          className="card-text text-left"
        >
          great hike...
        </p>
      </p>
    </div>
  </div>,
  " ",
]
`;
